### DISCUSSION TITLE
What is the difference between decorateRequest and @fastify/request-context

### DISCUSSION THREAD

**Discussion by joshuaavalon** [fastify/fastify#4202] [Q&A] [answered]
As title, is there any difference between between decorateRequest and @fastify/request-context?

Both of them seems to able to create request-scoped variables.

**Comment by mcollina** [comment, marked as answer]
The first has zero overhead, and the latter cost about 15% of throughput.

**Reply by DibyodyutiMondal** [reply]
Follow up:
Given that `decorateRequest` is more performant and requires less dependencies, is there any use case where `@fastify/request-context` is better than the `decorateRequest` method?
